java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java

```diff
@@ -85,7 +87,7 @@ public HttpEventResponse handle(HttpEventRequest request) {
         } else {
             builder.statusCode(HttpStatus.SC_FORBIDDEN);
             builder.header(ResponseHeader.ERROR.getHttpHeader(), ErrorCauses.BLOCKED_BY_RULES.name());
-            log.warning(String.format("%s. Blocked call by rules %s", callLog, objectMapper.writeValueAsString(rules.getAllowedEndpointRegexes())));
+            log.warning(String.format("%s. Blocked call by rules %s", callLog, objectMapper.writeValueAsString(rules)));
```

Comment:
we return the sha of the rules as a header, so we can at least see if they're running a known version or not. 

returning encoded rules themselves feels like it may open security concerns. Eg, someone opened some API endpoint for some reason, but didn't robustly test the behavior.  From SHA alone, you don't necessarily know this; but from SHA you do know which well-known rule set they're using.